src/PostgREST/Main.hs

```diff
@@ -61,6 +68,9 @@ main = do
   pool :: H.Pool P.Postgres
           <- H.acquirePool pgSettings poolSettings
 
+  Right supported <- H.session pool isServerVersionSupported
```

Comment:
What do you think of something like:
```haskell
  resOrError <- H.session pool isServerVersionSupported
  either (fail . show) (\supported -> unless supported $ fail "Cannot run in this PostgreSQL version, PostgREST needs at least 9.2.0") resOrError
```
In this fashion we exit printing the CxError everytime the return is a ```Left```